src/main/java/UseCases/CookbookUseCase.java

```diff
@@ -0,0 +1,51 @@
+package UseCases;
+
+import Commands.Implementations.CommandImpl;
+import Entities.Recipe;
+import Entities.Tag;
+import Storages.Implementations.RecipeStorageImpl;
+import Storages.Implementations.TagStorageImpl;
+
+import java.util.*;
+
+public class CookbookUseCase {
+    /**
+     * The cookbook displays recipes and shows recipes based on tags inputed into command
+     */
+    final RecipeStorageImpl recipeStorage;
+    final TagStorageImpl tagStorage;
+
+    public CookbookUseCase(RecipeStorageImpl recipes, TagStorageImpl tags) {
+        this.recipeStorage = recipes;
+        this.tagStorage = tags;
+    }
+
+    public RecipeResponseImpl run(CommandImpl command) {
+        List<Recipe> recipesMatched = new ArrayList<>();
+
+        if (!Objects.equals(command.get("FindAllRecipes"), "")) {
+            recipesMatched = new ArrayList<>(this.recipeStorage.recipes());
+
+        // If the command FindRecipesByTags was run with given tag
+        // This command allows us to find all the recipes that do not contain the given tags
+        } else if (!Objects.equals(command.get("FindRecipesByTags"), "")) {
```

Comment:
This statement can be simplified. I personally am a huge fan of the "return early" approach. So instead of having this else if, you can have an early return in this first `if`. Only one branch of the if can run and when the first does all that is done after is returning the response. So return it within this if.

Then, we don't need `else if` just another `if` statement after. The idea is that generally `else`  is not necessary and only complicates things (albeit not much).